clients/src/main/java/org/apache/kafka/common/security/scram/ScramExtensionsCallback.java

```diff
@@ -14,33 +14,35 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.apache.kafka.common.security.scram;
 
+import org.apache.kafka.common.security.auth.SaslExtensionsCallback;
+
 import javax.security.auth.callback.Callback;
 import java.util.Collections;
 import java.util.Map;
 
+
 /**
- * Optional callback used for SCRAM mechanisms if any extensions need to be set
- * in the SASL/SCRAM exchange.
+ * @deprecated As of 2.1.0, use {@link SaslExtensionsCallback} instead. This will be removed in a future major release.
  */
+@Deprecated
 public class ScramExtensionsCallback implements Callback {
```

Comment:
I thought we weren't supporting `SaslExtensionsCallback` for SCRAM. We should either update `ScramSaslClient` to process `SaslExtensions` and add tests for that  or not deprecate this now. In any case, I am not sure why the javadoc was removed.

Reply:
Removing the javadoc was a mistake.
I will look into making `ScramSaslClient` work with `SaslExtensions`. Last time I tried some tests kept failing for a reason I could not debug even after significant effort. If it happens to be the case again I'll simply remove the deprecated tag